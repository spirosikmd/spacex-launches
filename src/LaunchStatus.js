import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import GroupWorkRoundedIcon from '@material-ui/icons/GroupWorkRounded';
import MoreHorizRoundedIcon from '@material-ui/icons/MoreHorizRounded';

const styles = theme => ({
  status: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: '50%',
    padding: theme.spacing.unit / 2,
    zIndex: '1',
  },
  statusIcon: {
    fontSize: 16,
  },
  upcoming: {
    color: theme.palette.grey['500'],
  },
  success: {
    color: theme.palette.success.main,
  },
  fail: {
    color: theme.palette.error.main,
  },
  inProgress: {
    color: theme.palette.grey['500'],
  },
});

const LaunchStatus = ({
  classes,
  isSuccessful,
  isFailed,
  isUpcoming,
  isInProgress,
}) => {
  const statusClassName = classnames(classes.status, {
    [classes.success]: isSuccessful,
    [classes.fail]: isFailed,
    [classes.upcoming]: isUpcoming,
    [classes.inProgress]: isInProgress,
  });

  return (
    <Paper className={statusClassName} elevation={1}>
      {isInProgress ? (
        <MoreHorizRoundedIcon className={classes.statusIcon} />
      ) : (
        <GroupWorkRoundedIcon className={classes.statusIcon} />
      )}
    </Paper>
  );
};

LaunchStatus.propTypes = {
  classes: PropTypes.object.isRequired,
  isSuccessful: PropTypes.bool.isRequired,
  isFailed: PropTypes.bool.isRequired,
  isUpcoming: PropTypes.bool.isRequired,
  isInProgress: PropTypes.bool.isRequired,
};

export default withStyles(styles)(React.memo(LaunchStatus));
