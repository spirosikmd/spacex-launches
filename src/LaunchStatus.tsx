import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import {
  withStyles,
  Theme,
  createStyles,
  WithStyles,
} from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import GroupWorkRoundedIcon from '@material-ui/icons/GroupWorkRounded';
import MoreHorizRoundedIcon from '@material-ui/icons/MoreHorizRounded';

const styles = (theme: Theme) =>
  createStyles({
    status: {
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      borderRadius: '50%',
      padding: theme.spacing.unit / 2,
      zIndex: 1,
    },
    statusIcon: {
      fontSize: 16,
    },
    upcoming: {
      color: theme.palette.grey['500'],
    },
    success: {
      color: theme.success.main,
    },
    fail: {
      color: theme.palette.error.main,
    },
    inProgress: {
      color: theme.palette.grey['500'],
    },
  });

interface LaunchStatusProps extends WithStyles<typeof styles> {
  isSuccessful: boolean;
  isFailed: boolean;
  isUpcoming: boolean;
  isInProgress: boolean;
}

const LaunchStatus = ({
  classes,
  isSuccessful,
  isFailed,
  isUpcoming,
  isInProgress,
}: LaunchStatusProps) => {
  const statusClassName = classnames(classes.status, {
    [classes.success]: isSuccessful,
    [classes.fail]: isFailed,
    [classes.upcoming]: isUpcoming,
    [classes.inProgress]: isInProgress,
  });

  return (
    <Paper className={statusClassName} elevation={1}>
      {isInProgress ? (
        <MoreHorizRoundedIcon className={classes.statusIcon} />
      ) : (
        <GroupWorkRoundedIcon className={classes.statusIcon} />
      )}
    </Paper>
  );
};

LaunchStatus.propTypes = {
  classes: PropTypes.object.isRequired,
  isSuccessful: PropTypes.bool.isRequired,
  isFailed: PropTypes.bool.isRequired,
  isUpcoming: PropTypes.bool.isRequired,
  isInProgress: PropTypes.bool.isRequired,
};

export default withStyles(styles)(React.memo(LaunchStatus));
