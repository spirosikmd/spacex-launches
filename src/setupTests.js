import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createSerializer } from 'enzyme-to-json';
import { unwrap } from '@material-ui/core/test-utils';

configure({ adapter: new Adapter() });

expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

global.shallow = (Component, props = {}) => {
  const ComponentWithoutStyles = unwrap(Component);
  return shallow(<ComponentWithoutStyles {...props} />);
};

global.mount = (Component, props = {}) => {
  const ComponentWithoutStyles = unwrap(Component);
  return mount(<ComponentWithoutStyles {...props} />);
};

global.Date = jest.fn().mockImplementation(() => {
  return {
    toUTCString: jest.fn().mockReturnValue('Fri, 11 Jan 2019 15:31:00 GMT'),
    getDate: jest.fn().mockReturnValue(3),
    getMonth: jest.fn().mockReturnValue(10),
    getFullYear: jest.fn().mockReturnValue(2019),
  };
});

// https://github.com/airbnb/enzyme/issues/1875
jest.mock('react', () => {
  const r = jest.requireActual('react');
  return { ...r, memo: x => x };
});
