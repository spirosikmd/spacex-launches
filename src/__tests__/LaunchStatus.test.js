import React from 'react';
import LaunchStatus from '../LaunchStatus';
import { render } from '../setupTests';

describe('LaunchStatus', () => {
  let props;

  beforeEach(() => {
    props = {
      isSuccessful: true,
      isFailed: false,
      isUpcoming: false,
      isInProgress: false,
    };
  });

  it('renders successful', () => {
    const { asFragment } = render(<LaunchStatus {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders failed', () => {
    props.isFailed = true;
    props.isSuccessful = false;
    const { asFragment } = render(<LaunchStatus {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders upcoming', () => {
    props.isUpcoming = true;
    props.isSuccessful = false;
    const { asFragment } = render(<LaunchStatus {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders in progress', () => {
    props.isInProgress = true;
    props.isSuccessful = false;
    const { asFragment } = render(<LaunchStatus {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
