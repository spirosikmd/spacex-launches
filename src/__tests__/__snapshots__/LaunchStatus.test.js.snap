// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchStatus renders failed 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation1 LaunchStatus-status-43 LaunchStatus-fail-47 MuiPaper-rounded"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root LaunchStatus-statusIcon-44"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM8 17.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5zM9.5 8c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8zm6.5 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`LaunchStatus renders in progress 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation1 LaunchStatus-status-127 LaunchStatus-inProgress-132 MuiPaper-rounded"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root LaunchStatus-statusIcon-128"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`LaunchStatus renders successful 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation1 LaunchStatus-status-1 LaunchStatus-success-4 MuiPaper-rounded"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root LaunchStatus-statusIcon-2"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM8 17.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5zM9.5 8c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8zm6.5 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`LaunchStatus renders upcoming 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation1 LaunchStatus-status-85 LaunchStatus-upcoming-87 MuiPaper-rounded"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root LaunchStatus-statusIcon-86"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM8 17.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5zM9.5 8c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8zm6.5 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
      />
    </svg>
  </div>
</DocumentFragment>
`;
